pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -65,19 +70,38 @@ def work(self):
                 candy = inventory.candies().get(pokemon.pokemon_id)
                 pokemon_to_be_evolved = pokemon_to_be_evolved + min(candy.quantity / (pokemon.evolution_cost - 1), filtered_dict[pokemon.pokemon_id])
 
-        if pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved:
+        self._print_check(pokemon_to_be_evolved)
+
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        if has_minimum_to_evolve:
             if self.use_lucky_egg:
                 self._use_lucky_egg()
             cache = {}
             for pokemon in filtered_list:
                 if pokemon.can_evolve_now():
                     self._execute_pokemon_evolve(pokemon, cache)
 
+    def _print_check(self, pokemon_to_be_evolved):
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        result_message = ("Gotta catch`em all!", "Gotta evolv`em all!")[has_minimum_to_evolve]
+        self.emit_event(
+            'pokemon_evolve_check',
+            formatted='Checking... Has {has}, needs {needs}. {message}',
+            data={
+                'has': pokemon_to_be_evolved,
+                'needs': self.min_pokemon_to_be_evolved,
+                'message': result_message
+            }
+        )
+
     def _should_run(self):
         if not self.evolve_list or self.evolve_list[0] == 'none':
             return False
-        return True
-    
+        return self.next_update is None or datetime.now() >= self.next_update
```

Comment:
It seems like the duration that user set now affects how frequent evolve pokemon task is ran. I think only the logging should be reduced, but not the task itself. 

For example, if I set `min_pokemon_to_be_evolved` to 80, there's a chance that I might be evolving more than 80 pokemon due to evolve pokemon not running when I have exactly 80 pokemon.

While evolving more pokemon than the quota is not a big deal for me, I'm not sure if that applies to everyone else. Also if user accidentally sets `interval` to a huge number, that might be a bigger issue.